File: 4-median-of-two-sorted-arrays/4-median-of-two-sorted-arrays.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        merged_list = []
        i = 0
        j = 0
        
        total_len = len(nums1) + len(nums2)
        
        even = True
        if total_len % 2 != 0:
            even = False
        
        med_index = 0
        for _ in range(total_len):
            if i == len(nums1):
                for k in range(j, len(nums2)):
                    merged_list.append(nums2[k])
                    if med_index == total_len//2:
                        if not even:
                            return merged_list[med_index]
                        else:
                            return (merged_list[med_index] + merged_list[med_index-1])/2
                    med_index += 1
                break;
            if j == len(nums2):
                for k in range(i, len(nums1)):
                    merged_list.append(nums1[k])
                    if med_index == total_len//2:
                        if not even:
                            return merged_list[med_index]
                        else:
                            return (merged_list[med_index] + merged_list[med_index-1])/2
                    med_index += 1
                break;
            
            if nums1[i] < nums2[j]:
                merged_list.append(nums1[i])
                i+= 1
                
            else:
                merged_list.append(nums2[j])
                j+= 1
            
            if med_index == total_len//2:
                if not even:
                    return merged_list[med_index]
                else:
                    return (merged_list[med_index] + merged_list[med_index-1])/2
            med_index += 1
                    
        return -1 
        

File: 4-median-of-two-sorted-arrays/4-median-of-two-sorted-arrays.go
func findMedianSortedArrays(nums1 []int, nums2 []int) float64 {

    totalLength := len(nums1) + len(nums2) 
    mergedArray := make([]int, totalLength)
    var idx1, idx2 int
    for i := 0; i < totalLength/2 + 1; i++ {    
        var value int
        
        if idx1 < len(nums1) && idx2 < len(nums2) {
            if nums1[idx1] <= nums2[idx2] {
                value = nums1[idx1]
                idx1++
            } else {
                value = nums2[idx2]
                idx2++
            }     
        } else if idx1 < len(nums1) {
            value = nums1[idx1]
            idx1++
        } else {
            value = nums2[idx2]
            idx2++
        }
        
        mergedArray[i] = value 
    } 
    
    if totalLength % 2 == 0 {
        x1 := float64(mergedArray[totalLength/2 - 1])
        x2 := float64(mergedArray[totalLength/2])
        return (x1+x2)/2
    } else {
        return float64(mergedArray[totalLength/2])
    }
}